### DISCUSSION TITLE
Abstract title in DOCX output

### DISCUSSION THREAD

**Discussion by wuffi** [jgm/pandoc#9736] [Q&A] [answered]
Upon generating updated DOCX output with `pandoc` (version 3.1.3) from an older Markdown file, I realized the addition of the title "Abstract" to the abstract, which was not the case before.
This behaviour most have been implemented somewhere between version 3.1.1 and 3.1.3.
The problem in version 3.1.3 was that the style of the "Abstract" title in DOCX output was "normal", which means that any modifications to it in a custom template affect the style of many other parts.
Is there a way to modify or assign a custom style to the "Abstract" title? I'm not sure whether I can simply update, because version 3.1.3 is the version which ships with current versions of RStudio.

**Comment by jgm** [comment]
With the current version of pandoc, the word "Abstract" gets the style "Abstract Title", so you can certainly adjust it.

**Reply by wuffi** [reply]
Thanks, this is a good point:
It is defined in `data/docx/word/styles.xml`, which is different in older versions of `pandoc`.  
Is there a way hand over a newer version when invoking `pandoc` on the command line?

**Reply by jgm** [reply]
You could try generating a reference.docx using the latest pandoc and using it with the older version, but that is likely not to work perfectly.

**Reply by wuffi** [reply]
I like the possibility of using Lua filters a lot and spent some time in trying write a filter for modifying the abstract title in `DOCX` output.
While it's easy to access metadata elements as described in the documentation, I found no way to filter for the word 'Abstract' despite using the `logging` module which is usually very helpful.

**Reply by tarleb** [reply]
Note that the "Abstract" title is added after filters have run, namely by docx writer, so it's not available to filters. The writer uses the `abstract-title` metadata field, so you can change that title with something like

``` yaml
---
abstract-title: ABSTRACT
---
```

Mentioning only for completeness: The default value used for the abstract title is language dependent. I.e., setting `lang: fr` will change the value as well.

**Comment by wuffi** [comment]
It's great to see *Pandoc* evolve and many useful features added.
In case of modifications which aren't reflected in the AST like the one for the abstract title, using a custom writer stored by users alongside the document I want to convert instead of a filter is probably the way to go.
I will carefully look into <https://pandoc.org/custom-writers.html> on how to do this, because I'm only familiar with using filters so far.
If you have quick hint how I could use the older version of the docx writer as a custom writer, I would be very grateful.

**Reply by tarleb** [reply]
I'd advise against writing a custom writer for docx: it's probably the most complex format supported by pandoc, with the writer containing the most code of all writers. The reference material for docx has more than 1100 pages.

Custom writers are a good tool for simpler formats, but not for docx.

**Reply by wuffi** [reply, marked as answer]
Thanks for all your help, which finally enabled me to pin down the problem.

By carefully looking at `src/Text/Pandoc/Writers/Docx.hs` for my version of *Pandoc* (3.1.3), I finally realized that everything is well coded.  

There is a unexpected/surprising thing about using a template/reference *docx* file, which now makes me understand the first answer by @jgm who was of course absolutely right:  
Since there hadn't been a *Abstract Title* style in my older template/reference  *docx* file, the output by the newer `pandoc` version would also not have nor add an *Abstract Title* style. Output generated with a newer `pandoc` **without using a template/reference *docx* file** adds the abstract title "Abstract" **with the style *Abstract Title* **.  
I learn that I should also generate the template/reference *docx* file again to avoid causing you work.
Maybe it's worth to add this conclusion somewhere to the manual so that also other profit from it: *docx* output is unfortunately complicated, but nonetheless widely used.
